### PR TITLE
Update dialects

### PR DESCRIPTION
**Pull request by webmiche**
This updates our mirrored dialects. In partiuclar, moves std to func, and moves the funcOp from builtin to func.

### CODE REVIEW

**Comment by webmiche** [comment]
I had some weird behaviour: After installing the rewritten dialects with `pip install .`, most tests would still work, i.e., `std.return` would still be parsed, the imports of `std` still work...
On the CI however, these cases failed. I am uncertain how my local environment differs from the CI. Could it be that pip does not clean up properly when reinstalling something directly?

**Comment by tobiasgrosser** [comment]
Nice! Good from my side. Maybe @math-fehr has some further input.

File: README.md [hunk 1]
```diff
@@ -24,6 +22,17 @@ pytest
 lit tests/filecheck
 ```
 
+## Generating executables through MLIR
+
+xDSL can generate executables using MLIR as the backend. To use this functionality, make sure to install the [MLIR Python Bindings](https://mlir.llvm.org/docs/Bindings/Python/). Given an input file `input.xdsl`, that contains IR with only the mirrored dialects found in `src/xdsl/dialects` (arith, memref, func, cf, scf, and builtin), run:
+
+```
+mlir-opt --convert-scf-to-cf --convert-cf-to-llvm --convert-func-to-llvm --convert-arith-to-llvm --convert-memref-to-llvm --reconcile-unrealized-casts *input.xdsl* | mlir-translate --mlir-to-llvmir > tmp.ll
```
Comment (math-fehr) [inline README.md:30]: How can we directly use an `input.xdsl` file in MLIR?
Don't we need to use mlir_converter or something like that?

**Review by math-fehr — APPROVED** [review]
Besides my documentation comment, looks good to me!

File: README.md (hunk 1 above)
Comment (webmiche) [reply README.md:30]: Yes, you are right, it should actually be passed through the mlir_converter aswell. I will think about how to properly integrate the tooling from the ChocoPy repo into xdsl. (I will probably add `.mlir` as a target to xdsl_opt, similar to how we did in choco-opt)

**Review by Dinistro — APPROVED** [review]
This looks good to me. 
There are some ugly things in the files touched by this PR, but changing this is not related to it.

**Pull request merged**